src/data_api/events.se

```diff
@@ -1,346 +1,420 @@
-
-# Keeps track of number of events created in the past 24 hours
-data past24Hours[]
+inset('../macros/safeMath.sem')
+inset('../macros/unitConversions.sem')
 
 data controller
 
-event logOutcome(event:indexed, outcome)
+# tokens[branch][outcome][tokenHolderAddress]
+# tokenSupply[branch][outcome]
+# allOutcomesTokenSupply[branch]
+data events[](branch, endTime, description, automatedReporterAddress, automatedReportDueTimestamp, reportingWindow, finalOutcome, tentativeOutcome, automatedReporterOutcome, automatedReporterDisputeBondHolder, limitedReportersOutcome, limitedReportersDisputeBondHolder, allReportersOutcome, allReportersDisputeBondHolder, tokens[][][2**160], tokenSupply[][],  allOutcomesTokenSupply[])
+# spenders[tokenHolderAddress][spenderAddress]
+data spenders[2**160][2**160]
+data nextEventId
+### TODO FIX where are the setters for numberOfLimited and numberOfAll
+# takes branch, reportingWindow as index
+data numberOfLimitedReporterEvents[][]
+# takes branch, reportingWindow as index
+data numberOfAllReporterEvents[][]
+data automatedReporterDisputeBondAmount
+data limitedReportersDisputeBondAmount
+data allReportersDisputeBondAmount
+
+event Transfer(eventId, branch:uint256:indexed, outcome, from:address:indexed, to:address:indexed, attorep:uint256)
+event Approval(owner:address:indexed, spender:address:indexed, attorep:uint256)
+
 
 def init():
     self.controller = 0x0
-
-def getForkResolveAddress(event):
-    return(self.Events[event].forkResolveAddress)
-
-def getCreationTime(event):
-    return(self.Events[event].creationTime)
-
-def setCreationTime(event):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].creationTime = block.timestamp
+    # TODO: figure out approprate values for these
+    self.automatedReporterDisputeBondAmount = 11 * 10**20
+    self.limitedReportersDisputeBondAmount = 11 * 10**21
+    self.allReportersDisputeBondAmount = 11 * 10**22
+    self.nextEventId = 1
+
+def any():
+    refund()
+
+def createEvent(branch, endTime, description, automatedReporterAddress):
+    # FIXME: assert the user sent enough fees and validity bond
+    # FIXME: assert not a duplicate event. CONSIDER: is this actually necessary?
+    # FIXME: add event to tally of total events in reporting window, also be sure to move this anytime the event changes reporting windows
+    eventId = self.nextEventId
+    self.nextEventId += 1
+    self.events[eventId].branch = branch
+    self.events[eventId].endTime = endTime
+    self.events[eventId].description = description
+    self.events[eventId].automatedReporterAddress = automatedReporterAddress
+    earliestReportingTime = endTime
+    if (automatedReporterAddress != 0):
+        earliestReportingTime = endTime + 6 * SECONDS_PER_DAY
+    self.events[eventId].reportingWindow = self.controller.lookup('reporting').getReportingWindowForTimestamp(earliestReportingTime) + 1
+    return(eventId)
+
+def report(eventId, outcome, attorepStake):
```

Comment:
Should also assert something like 

```
    max = EVENTS.getMaxValue(event)
    min = EVENTS.getMinValue(event)
    if(outcome < min):
        outcome = min
    elif(outcome > max):
        outcome = max
```

Or alternatively throw if the outcome is > max or < min